File: 2022/HahaKoreaShop/app/src/main/java/com/ncggloval/hahakoreashop/util/EtcUtil.java
package com.ncggloval.hahakoreashop.util;

import java.util.Random;

public class EtcUtil {
    public static String getRandomKey(int length) {
        StringBuffer temp = new StringBuffer();
        Random rnd = new Random();
        for (int i = 0; i < length; i++) {
            int rIndex = rnd.nextInt(3);
            switch (rIndex) {
                case 0:
                    // a-z
                    temp.append((char) ((int) (rnd.nextInt(26)) + 97));
                    break;
                case 1:
                    // A-Z
                    temp.append((char) ((int) (rnd.nextInt(26)) + 65));
                    break;
                case 2:
                    // 0-9
                    temp.append((rnd.nextInt(10)));
                    break;
            }
        }
        return temp.toString();
    }

    public static String EscapeJavaScriptFunctionParameter(String input) {
        String result = input;
        if(result.length() > 0) {
            result.replace("\\", "\\\\").replace("'", "\\'").replace("\b", "\\b").replace("\n", "\\n").replace("\t", "\\t").replace("\f", "\\f").replace("\r", "\\r");
        }

        return result;
    }
}

File: 2022/HahaKoreaShop/app/src/main/java/com/ncggloval/hahakoreashop/util/EtcUtil.kt
package com.ncggloval.hahakoreashop.util

import java.util.*

object EtcUtil {
    fun getRandomKey(length: Int): String {
        val temp = StringBuffer()
        val rnd = Random()
        for (i in 0 until length) {
            val rIndex = rnd.nextInt(3)
            when (rIndex) {
                0 ->                     // a-z
                    temp.append((rnd.nextInt(26) + 97).toChar())
                1 ->                     // A-Z
                    temp.append((rnd.nextInt(26) + 65).toChar())
                2 ->                     // 0-9
                    temp.append(rnd.nextInt(10))
            }
        }
        return temp.toString()
    }

    fun EscapeJavaScriptFunctionParameter(input: String): String {
        if (input.length > 0) {
            input.replace("\\", "\\\\").replace("'", "\\'").replace("\b", "\\b")
                .replace("\n", "\\n").replace("\t", "\\t").replace("\u000c", "\\u000c")
                .replace("\r", "\\r")
        }
        return input
    }
}